first_try

Felicuss opened the pull request.

NemirovichMark committed:
Update README.md

Felicuss committed:
First Try

NemirovichMark commented:
Здравствуйте, Ярослав.

Во 2м номере переименовать JuniorSkyJumping в JuniorSkiJumping
И Standart в Standard

В 3м номере инициализировать в любом случае. Нельзя null вместо экземпляра возвращать.

[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19270549/Lab_7_log_Purple.txt)

Felicuss committed:
try 2

Задание 1: Ничего не менял, т.к тесты не проходят по сути из-за незначительного округления (если надо будет исправить - попытаюсь найти причину, почему так происходит, скорее всего из-за умножения, а потом суммирования)
Задание 2: Исправил названия
Задание 3: Попытался исправить, но, к сожалению, некоторые ошибка не понял
Задание 4: Изменил метод Shuffle, теперь если у самого быстрого мужчины время лучше, чем у самой быстрой женщины, то начинаем с мужчины и наоборот
Задание 5: Исправил неправильный метод CountVotes, который был неправильным еще с прошлой лабораторной

Felicuss commented:
Исправил!

NemirovichMark commented:
Почти.

В 3м номере вы забираете весь массив, а нужно только первые 7 элементов
                _moods = new double[moods.Length];
                Array.Copy(moods, _moods, _moods.Length);

[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19273172/Lab_7_log_Purple.txt)

Felicuss committed:
try 3 (please last)

1 номер: оставил нетронутым, так как ошибка в округлениях (если надо исправить - буду разбираться в чем причина)
3 номер исправил: добавил в Evaluate проверку на добавление только 7 первых оценок в итоговый массив, а также в конструкторе сделал проверку на добавление лишь первых 7 элементов массива в итоговый
Надеюсь, теперь все нормально :)

Felicuss commented:
Исправил!

NemirovichMark commented:
Здравствуйте, Ярослав.
Отлично! Загрузите файлы на Moodle, чтобы поставил баллы за защиту (если ещё не сделали этого).
